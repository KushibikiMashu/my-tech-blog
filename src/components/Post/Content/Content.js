// @flow strict
import React from 'react';
import styles from './Content.module.scss';

type Props = {
  body: string,
  title: string,
  toc: string,
};

const insertTocIntoBody = (toc: string, body: string): string => {
  const article: HTMLElement = document.createElement('article');
  article.innerHTML = body;

  const firstElement: ?Node = article.firstChild;
  if (firstElement === null || firstElement === undefined) {
    return body;
  }

  const div: HTMLDivElement = document.createElement('div');
  div.innerHTML = toc;
  // 目次にクラス名を追加する
  // div.classList.add("test");

  // 記事の2番目の位置に目次を追加する
  article.insertBefore(div, firstElement.nextSibling);
  return article.outerHTML;
};

const Content = ({ body, title, toc }: Props) => {
  const article = insertTocIntoBody(toc, body);
  return (
  <div className={styles['content']}>
    <h1 className={styles['content__title']}>{title}</h1>
    {/* <Meta date={date}/> */}
    <div className={styles['content__body']} dangerouslySetInnerHTML={{ __html: article }}/>
    <div className={styles['content__body']}>
      <p>Happy Coding 🎉</p>
    </div>
  </div>
  );
};

export default Content;
