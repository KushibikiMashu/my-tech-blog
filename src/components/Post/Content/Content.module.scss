@import '../../../assets/scss/variables';
@import '../../../assets/scss/mixins';

.content {
  max-width: $layout-post-single-width;
  padding: 0 15px;
  margin: 0 auto;

  &__breadcrumb {
    margin: 14px 0 7px;
    padding-left: 0;
    list-style: none;
    display: flex;
    flex-wrap: wrap;
    justify-content: center;
    font-size: 14px;
    color: #4A5568;

    li {
      @include margin-top(0);
      @include margin-bottom(0);
    }
  }

  &__title {
    font-size: $typographic-base-font-size * 1.4;
    max-width: $layout-post-width;
    margin-left: auto;
    margin-right: auto;
    font-weight: 600;
    text-align: center;
    @include line-height(1.5);
    @include margin-top(0);
    @include margin-bottom(0);
  }

  &__body {
    & p {
      font-size: $typographic-base-font-size * 1;
      @include line-height(1.1);
    }

    & blockquote {
      p {
        @include margin-bottom(0.25);
      }
    }

    &-toc {
      @include padding(0.1, 0.3, 0.1, 0);
      @include line-height(1.25);
      background-color: #F7FAFC;
      border-radius: 8px;

      ul {

        li {
          font-size: $typographic-base-font-size * 1;
          @include margin-top(0.25);
          @include margin-bottom(0.25);

          ul {
            margin-top: 0;
            margin-bottom: 0;
            list-style: circle;
            padding-inline-start: 20px;

            ul {
              list-style: disc;
            }

          }

          p {
            margin: 0;
            font-size: $typographic-base-font-size * 1;
          }
        }
      }
    }

    & h2 {
      @include margin-top(3);
      @include line-height(1.5);
      border-bottom: 1px solid #ddd;
    }

    & h3, h4 {
      @include line-height(1.5);
      @include margin-top(1.5);
    }


    & figure {
      @include margin-bottom(1);

      & blockquote {
        & p {
          max-width: $layout-post-width;
          font-size: $typographic-base-font-size * 1.3;
          margin-top: 0;
          @include margin-bottom(1);
        }

      }

    }

    & a {
      text-decoration: none
    }

    & * {
      max-width: $layout-post-width;
      margin-left: auto;
      margin-right: auto;

    }

    & img {
      max-width: 100%;
    }

  }

}

@include breakpoint-md {
  .content {
    padding: 0;

    &__breadcrumb {
      margin: 21px 0 7px;
    }

    &__body {
      font-size: $typographic-base-font-size * 1.125;
      @include margin-bottom(1.125);

      & p {
        font-size: $typographic-base-font-size * 1.125;
        @include margin-bottom(1.125);
        @include line-height(1.25);
      }

      &-toc {
        ul {
          li {
            @include margin-top(0.5);
            @include margin-bottom(0.5);

            ul {
              padding-inline-start: 40px;
            }
          }
        }
      }
    }

  }
}
