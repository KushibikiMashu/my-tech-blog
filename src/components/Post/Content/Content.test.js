// @flow strict
import React from 'react';
import renderer from 'react-test-renderer';
import Content from './Content';

describe('Content', () => {
  it('renders correctly', () => {
    const props = {
      title: 'test',
      body: '<p>test</p>',
      toc: '<ul>test</ul>',
      date: '2019-12-02T09:00:00.000Z',
      updatedAt: '2019-12-01T09:00:00.000Z',
    };

    const tree = renderer.create(<Content {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
