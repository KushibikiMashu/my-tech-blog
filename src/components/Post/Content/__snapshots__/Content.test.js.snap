// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content renders correctly 1`] = `
<div
  className="content"
>
  <h1
    className="content__title"
  >
    test
  </h1>
  <div
    className="content__body"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<article><p>test</p><div><ul>test</ul></div></article>",
      }
    }
  />
  <div
    className="content__body"
  >
    <p>
      Happy Coding ðŸŽ‰
    </p>
  </div>
</div>
`;
